src/mcu_stm8s.c

```diff
@@ -255,6 +255,9 @@ static void UART_setup(void)
              UART2_PARITY_NO,
              UART2_SYNCMODE_CLOCK_DISABLE,
              UART2_MODE_TXRX_ENABLE);
+  
+  UART2_ITConfig(UART2_IT_RXNE_OR, ENABLE); 
+  enableInterrupts();
```

Comment:
Interrupts are to be enabled only after all peripherals are initialized - enableInterrupts() to be called from main only after all the inits are done, otherwise chaos may ensue!